packages/melos/lib/src/package.dart

```diff
@@ -366,6 +366,23 @@ class PackageFilters {
     );
   }
 
+  PackageFilters copyWithUpdatedScope(List<Glob> scope) {
```

Comment:
@spydon, sounds reasonable!
I was also a bit surprised not to see `copyWith` method, but initially decided to stick to the code style in the repository. 
I just replaced it with a more reusable `copyWith` in https://github.com/invertase/melos/pull/640/commits/769a3ab2ae968db405b5ef1fcf190ed58eccf6bf, implemented it using the `ValueProvider` function so that the nullable values can be passed as well.

I'd love to refactor the `copyWithDiff` and `copyWithDiff`, but I'd rather do it in a separate PR to keep the PRs topic focused.

Reply:
> Since only four of the values are nullable only those ones need the ValueProvider

That's true, I valued consistency over redundancy for that particular case, but can easily narrow it down only to the 4 values.

Reply:
Seemingly the default for `copyWith` methods seem to be to ignore null arguments, so maybe we could just skip the `ValueProvider` all together.
Example:
https://api.flutter.dev/flutter/painting/TextStyle/copyWith.html#source

Reply:
@spydon, I must be missing then how we would be able to cover [cases like this](https://github.com/search?q=repo%3Ainvertase%2Fmelos+packageFilters%3F.copyWithDiff%28null%29&type=code#:~:text=packageFilters%3A-,packageFilters%3F.copyWithDiff(null),-%2C) if we just ignore the null arguments for parameters that may be needed to be overridden with `null` values.

Reply:
We're not of course, so we'd need to keep the `copyWithDiff` method. But I think it is quite rare that we need to override with `null`, so as a trade-off for ergonomic reasons I would have the method like in Flutter's `copyWith` methods.

I wonder how freezed implemented this, because it used to (https://github.com/rrousselGit/freezed/issues/906) be able to receive null arguments and determine whether they were passed in explicitly or not.

Reply:
@spydon, we may get rid of using the `ValueProvider` here, but then we'd not be able to get rid of the thing you mentioned in the first comment, i.e. `copyWithDiff` will exist next to `copyWith` which is weird IMO.
> Seems like we're getting more and more copyWith methods

So, would you like me to remove the possibility of setting nulls with `copyWith`?

Reply:
> copyWithDiff will exist next to copyWith which is weird IMO.

It is a bit weird, but I think it is worth the trade-off until the day we get more than a single place passing in null, I think it's quite a YAGNI situation. We could name it something differently and not pass in anything, like `PackageFilters resetDiff()` to make it more clear how it is different from `copyWith`.

Reply:
I see your point @spydon, that's fine by me. Could you give the PR a re-review taking https://github.com/invertase/melos/pull/640/commits/bcc1379e42916e279d6527b8618505782c7db3cc into account?